collect-user-data

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env bash
```

Comment:
I've tried to follow the "node if you can, /bin/sh with `ash` extensions if you have to (e.g. boot-time on the OS images), and always avoid bash-isms" rule, but IMO there are some trivial/simple tasks where scripting it with /bin/sh is fine. Judging wether it's simple enough is probably a subjective call worth talking about in general, outside of the context of this review, but in the past I've found it easy for shell scripts to grow slowly feature-wise and leave me wishing I'd started in javascript with access to a bigger module library and a richer set of primitive data structures.

Not super-related, but Google has a shell style guide that might be an interesting reference: https://google.github.io/styleguide/shell.xml